sdk/identity/azure-identity/azure/identity/_internal/aad_client_base.py

```diff
@@ -310,6 +342,17 @@ def _post(self, data: Dict, **kwargs: Any) -> HttpRequest:
         return HttpRequest("POST", url, data=data, headers={"Content-Type": "application/x-www-form-urlencoded"})
 
 
+def _merge_claims_challenge_and_capabilities(capabilities, claims_challenge):
+    # Represent capabilities as {"access_token": {"xms_cc": {"values": capabilities}}}
+    # and then merge/add it into incoming claims
+    if not capabilities:
+        return claims_challenge
+    claims_dict = json.loads(claims_challenge) if claims_challenge else {}
+    for key in ["access_token"]:
```

Comment:
Why?

Reply:
This was borrowed from the msal [implementation](https://github.com/AzureAD/microsoft-authentication-library-for-python/blob/dev/msal/application.py#L49) for handling claims + capabilities. Seems like there is the potential for adding the xms_cc value for "id_token", hence the loop.